### ISSUE TITLE
dev-apprenticeship: auto-confidence from feedback signal (Option B, follow-up to #100)

### ISSUE THREAD

**Issue by ylohnitram** [Replikanti/agentis-colonies#106]
## Context

#100 corrected the README's claim that "Agents that predict correctly gain confidence. Stale knowledge decays." That framing sold the system as a real apprenticeship — agents earn their way up based on predictive accuracy. In reality confidence is manually set via `agentis memo set`, and without this feature the observe → suggest → act gradient is, as #100 put it, "just UI around a manual kill switch".

This issue tracks **Option B** from #100: implement auto-confidence so the apprenticeship framing becomes real.

## The feedback signal

For each suggesting agent (confidence 0.6-0.84), on each tick:

1. **Record**: when an agent emits a suggestion (e.g. `labeler` suggests `["bug", "api"]` for issue #42), record `(agent, artifact_id, suggestion, emit_tick)` as a pending verdict.
2. **Evaluate on subsequent ticks**: check what the human actually did to that artifact. For `labeler`: which labels were applied? For `approval_decider`: was the MR approved, rejected, or merged as-is? For `prioritizer`: what priority was set?
3. **Score**:
   - Exact match → `confidence += learning_rate.match` (default 0.02)
   - Partial / semantically close → `confidence += learning_rate.partial` (default 0.005)
   - Mismatch → `confidence -= learning_rate.mismatch` (default 0.01)
   - No human action after `feedback.timeout_ticks` (default 1440 = 1 day) → expire without scoring
4. **Cap**: stop auto-promoting at 0.85 — require explicit operator bump for autonomy. The system earns *suggestion trust*, not *autonomous trust*.

## Per-agent feedback implementation

Each agent type needs its own matcher because the "what did the human do" signal is different:

| Agent | Signal |
|-------|--------|
| `labeler` | compare suggested label set to the labels actually on the issue at evaluation time |
| `prioritizer` | compare suggested priority label to the one actually set |
| `issue_creator` | did the issue get closed as duplicate, marked wontfix, or triaged? |
| `style_reviewer`, `logic_reviewer`, `security_reviewer`, `test_reviewer` | did the MR get merged with or without follow-up commits addressing the flagged concerns? |
| `approval_decider` | did the human approve/reject the same way? |
| `scope_estimator`, `risk_assessor` | was the MR size / risk close to the estimate? |
| `task_decomposer` | did the work actually follow the proposed breakdown? |
| `plan_reviewer` | did the plan get followed as written or revised? |
| `code_writer`, `test_writer`, `refactorer` | did the generated code survive review? CI result is the quick signal |
| `commit_composer` | did the commit message survive or get amended? |
| `ship_decider`, `release_checker` | was the release shipped or aborted after the decision? |
| `changelog_writer`, `version_bumper` | was the output accepted or rewritten? |

## Persistence

Store pending verdicts in `.agentis/memo/<agent>:pending_verdicts` as JSON array or similar. Evaluate on each tick, aging out expired entries. The router already checks `.agentis/memo` each tick; adding one memo key per agent is no cost surprise.

## Knobs to expose

```toml
[feedback]
enabled = true               # default false initially, gated opt-in
match_rate = 0.02
partial_rate = 0.005
mismatch_rate = 0.01
timeout_ticks = 1440         # 1 day at 60 s ticks
autonomy_cap = 0.85          # auto-promotion stops here
```

Visible in dashboard once implemented (maybe a "Learning rate last 24h" column).

## Scope / phasing

This is substantial work — 14 different per-agent matchers at minimum. Suggest phasing:

- **Phase 1**: one representative agent per colony (`labeler`, `style_reviewer`, `plan_reviewer`, `code_writer`, `version_bumper`). Prove the feedback-loop architecture end-to-end.
- **Phase 2**: remaining agents in each colony.
- **Phase 3**: dashboard visualization of learning rate and recent verdicts.

Each agent that ships Phase 1+ changes the honest README framing for that agent. Until then, the operator-controlled narrative from #100 stays.

## Acceptance (full feature)

- [ ] Feedback infrastructure: pending-verdict persistence, timeout handling, per-agent matcher interface
- [ ] Phase 1: 5 reference agents emitting verdicts and updating `<agent>:confidence`
- [ ] Dashboard shows learning rate / recent verdicts per agent
- [ ] Config knobs documented in README
- [ ] README updated to restore "agents gain confidence" narrative with caveat about the 0.85 autonomy cap
- [ ] Telemetry: `agentis remediation history` or similar shows confidence change events so the operator can audit what happened

## Related

- #100 (what this completes)
- #104 (operator-aware tagging — could feed into "only score verdicts on operator's own actions")
- #105 (dashboard confidence control — complements auto-confidence with manual override)
